generate_analysis_graphs.py

```diff
@@ -84,6 +85,84 @@
         individuals = TracedDataJsonIO.import_jsonl_to_traced_data_iterable(f)
     log.info(f"Loaded {len(individuals)} individuals")
 
+    log.info(f'Computing repeat and new participation per show ...')
+    # Computes the number of new and repeat consented individuals who participated in each radio show.
+    # Repeat participants are consented individuals who participated in previous shows prior to the target show.
+    # New participants are consented individuals who participated in target show but din't participate in previous shows.
+    repeat_new_participation_map = OrderedDict()  # of rqa_raw_field to participation metrics.
+
+    rqa_raw_fields =  [plan.raw_field for plan in PipelineConfiguration.RQA_CODING_PLANS]
+
+    for rqa_raw_field in rqa_raw_fields:
+        target_radio_show = rqa_raw_field  # radio show in which we are calculating repeat and new participation metrics for.
+
+        target_radio_show_participants = set()  # contains uids of individuals who participated in target radio show.
+        for ind in individuals:
+            if ind["consent_withdrawn"] == Codes.TRUE:
+                continue
+
+            if target_radio_show in ind:
+                target_radio_show_participants.add(ind['uid'])
+        log.debug(f'No. of uids in {target_radio_show} = {len(target_radio_show_participants)} ')
+
+        previous_radio_shows = []  # rqa_raw_fields of shows that aired before the target radio show.
+        for rqa_raw_field in rqa_raw_fields:
+            if rqa_raw_field == target_radio_show:
+                break
+
+            previous_radio_shows.append(rqa_raw_field)
+
+        previous_radio_shows_participants = set()  # uids of individuals who participated in previous radio shows.
+        for rqa_raw_field in previous_radio_shows:
+            for ind in individuals:
+                if ind["consent_withdrawn"] == Codes.TRUE:
+                    continue
+
+                if rqa_raw_field in ind:
+                    previous_radio_shows_participants.add(ind['uid'])
+        log.debug(f'No. of uids in {len(previous_radio_shows)} previous_radio_shows = {len(previous_radio_shows_participants)} ')
+
+        repeat_participants = set()  # uids of individuals who participated in target and previous shows.
+        new_participants = set()  # uids of individuals who participated in target show but din't participate in previous shows.
+        for uid in target_radio_show_participants:
+            if uid in previous_radio_shows_participants:
+                repeat_participants.add(uid)
+            else:
+                new_participants.add(uid)
```

Comment:
I think you can write this shorter and cleaner by saying something like:
`repeat_participants = target_radio_show_participants.intersect(previous_radio_show_participants)`

(or whatever python calls set intersection and union)

Reply:
Thanks for the tips have used .intersection() for repeat and .difference() for new participants